# Individual Plan - Stephanie Ye

In [77]:
library(tidyverse)

players_data <- read_csv("IPS/data/players.csv")

head(players_data)

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

In [78]:
sessions_data <- read_csv("IPS/data/sessions.csv")

head(sessions_data)

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


hashedEmail                                                     
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
  start_time       end_time         original_start_time original_end_time
1 30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2 17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3 25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4 25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5 25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12      
6 23/06/2024 15:08 23/06/2024 17:10 1.71916e+12         1.71916e+12

## Data description

There are 196 oberservation and 9 variables in players table, the variables are:
1. `experience`categorical, cdecribe the experience level of players.
2. `subscribe`logical, decribe whether the players subscribe the game or not.
3. `hashedEmail`categorical, decribe the hashing encoded email addresses of players.
4. `played_hours`numeric, decribe how long the players play the game.

There are 1535 oberservations and 5 variables in sessions table, the variables are:
1. `hashedEmail` categorical, describe the hashing encoded email addresses of players.
2. `start_time` and `end_time`categorical, describe when the players start or end playing the game.
3. `original_start_time` and `original_end_time` numerical, describe the UNIX timestamp recorded by system.

## potential issues

### players

In [79]:
players_data|>
group_by(gender) |>
summarize(count=n())

gender            count
1 Agender             2  
2 Female             37  
3 Male              124  
4 Non-binary         15  
5 Other               1  
6 Prefer not to say  11  
7 Two-Spirited        6

1. In the `hashedEmail` column, the data is unreadable which may be useless.
2. There may be some extreme value in `played_hours` which may affect the final results.
3. Some categories in `gender` may be too small, such as `Other`, `Two-Spirited`. Model may cannot show the patterns of these groups.

### Sessions

In [76]:
sessions_data |>
mutate(duration_mins = as.numeric(difftime(end_time, start_time, units = "mins"))) |>
select(start_time, end_time,duration_mins)

start_time       end_time         duration_mins
1    30/06/2024 18:12 30/06/2024 18:24 0            
2    17/06/2024 23:33 17/06/2024 23:46 0            
3    25/07/2024 17:34 25/07/2024 17:57 0            
4    25/07/2024 03:22 25/07/2024 03:58 0            
5    25/05/2024 16:01 25/05/2024 16:12 0            
6    23/06/2024 15:08 23/06/2024 17:10 0            
7    15/04/2024 07:12 15/04/2024 07:21 0            
8    21/09/2024 02:13 21/09/2024 02:30 0            
9    21/06/2024 02:31 21/06/2024 02:49 0            
10   16/05/2024 05:13 16/05/2024 05:52 0            
11   03/07/2024 01:31 03/07/2024 01:35 0            
12   30/07/2024 15:15 30/07/2024 15:43 0            
13   24/08/2024 02:32 24/08/2024 03:12 0            
14   18/08/2024 00:51 18/08/2024 03:15 0            
15   23/05/2024 18:12 23/05/2024 18:22 0            
16   11/07/2024 05:20 11/07/2024 05:57 0            
17   01/08/2024 23:12 01/08/2024 23:23 0            
18   08/08/2024 00:21 08/08/2024 01:35 0            
19   03/06/2024 00:22 03/06/2024 00:33 0            
20   31/07/2024 02:58 31/07/2024 03:21 0            
21   04/07/2024 02:25 04/07/2024 04:05 0            
22   07/05/2024 19:23 07/05/2024 19:34 0            
23   27/06/2024 21:40 27/06/2024 21:48 0            
24   30/05/2024 06:11 30/05/2024 06:57 0            
25   26/07/2024 23:09 26/07/2024 23:16 0            
26   09/07/2024 17:56 09/07/2024 18:02 0            
27   04/05/2024 21:41 04/05/2024 21:50 0            
28   26/06/2024 01:58 26/06/2024 03:35 0            
29   30/06/2024 16:21 30/06/2024 20:32 0            
30   22/08/2024 03:02 22/08/2024 03:49 0            
⋮    ⋮                ⋮                ⋮            
1506 08/06/2024 16:14 08/06/2024 20:25      0       
1507 02/06/2024 15:18 02/06/2024 16:53      0       
1508 22/05/2024 00:51 22/05/2024 01:22      0       
1509 29/08/2024 03:42 29/08/2024 04:11      0       
1510 03/08/2024 22:33 04/08/2024 00:51 527040       
1511 28/08/2024 19:00 28/08/2024 20:26      0       
1512 18/08/2024 21:54 19/08/2024 02:09 525600       
1513 23/05/2024 20:04 23/05/2024 20:20      0       
1514 22/07/2024 08:02 22/07/2024 08:07      0       
1515 02/06/2024 01:17 02/06/2024 01:21      0       
1516 29/04/2024 04:14 29/04/2024 05:34      0       
1517 27/04/2024 04:05 27/04/2024 05:48      0       
1518 30/07/2024 03:44 30/07/2024 03:54      0       
1519 28/05/2024 03:58 28/05/2024 04:04      0       
1520 27/08/2024 17:22 27/08/2024 17:31      0       
1521 12/06/2024 23:13 12/06/2024 23:32      0       
1522 10/09/2024 00:20 10/09/2024 04:34      0       
1523 04/08/2024 04:29 04/08/2024 08:01      0       
1524 09/08/2024 02:43 09/08/2024 03:14      0       
1525 23/05/2024 18:51 23/05/2024 18:58      0       
1526 25/06/2024 01:08 25/06/2024 02:18      0       
1527 19/05/2024 15:05 19/05/2024 17:02      0       
1528 05/07/2024 04:29 05/07/2024 04:45      0       
1529 22/06/2024 02:30 22/06/2024 04:22      0       
1530 28/08/2024 23:31 29/08/2024 00:15 525600       
1531 10/05/2024 23:01 10/05/2024 23:07      0       
1532 01/07/2024 04:08 01/07/2024 04:19      0       
1533 28/07/2024 15:36 28/07/2024 15:57      0       
1534 25/07/2024 06:15 25/07/2024 06:22      0       
1535 20/05/2024 02:26 20/05/2024 02:45      0

1. The `hashedEmail` is unreadable, which may be useless.
2. There are sessions with a duration of 0 minutes or with very long duration . Such cases might indicate logging errors or players disconnecting immediately. These values could affect the final results of prediction model.
3. The `start_time` and `end_time` may need to convert into another form in order to calculate more easily.